src/lib/utils/header.ts

```diff
@@ -105,41 +106,29 @@ function makeColumns(hostUrl, linkData, arrayDepth, promo): MegaMenuLinkObject {
     }
 
     // Generate promo block column if present
-    if (promo !== undefined || null) {
-      columns[columnNames[i]] = promo
+    if (promo && promo.id) {
+      columns[columnNames[i]] = queries.formatData(
+        'block_content--promo',
+        promo
+      )
```

Comment:
I find myself wondering if we still need a separate query file for block promos. Will we ever need to query this separately? Or should the formatting logic for promos be moved into this file?

Reply:
I am not 100% sure yet. These promo blocks are also included on the main benefits hubs pages in addition to the header megamenu. I think the safe route is to keep the formatting logic separate from this header file for now. 

The query may be redundant but the formatting will be used in both places

Reply:
Sounds fair!